src/pudl/transform/eia923.py

```diff
@@ -1223,3 +1223,96 @@ def _core_eia923__fuel_receipts_costs(
     frc_df = frc_df.dropna(subset=["energy_source_code"])
 
     return frc_df
+
+
+@asset
+def _core_eia923__fgd_operation_maintenance(
+    raw_eia923__fgd_operation_maintenance: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transforms the _core_eia923__fgd_operation_maintenance table.
+
+    Transformations include:
+
+    * Drop values with plant and boiler id values of NA.
+    * Replace . values with NA.
+    * Convert dollars to thousands of dollars.
+    * Fix datetimes for SO2 test dates.
+    * Ensure a unique primary key and drop some duplicated rows.
+
+    Args:
+        raw_eia923__fgd_operation_maintenance: The raw ``raw_eia923__fgd_operation_maintenance`` dataframe.
+
+    Returns:
+        Cleaned ``_core_eia923__fgd_operation_maintenance`` dataframe ready for harvesting.
+    """
+    fgd_df = raw_eia923__fgd_operation_maintenance
+
+    # Drop duplicate and empty rows to ensure the primary key is unique.
+    fgd_df = fgd_df.dropna(subset=["plant_id_eia", "so2_control_id_eia"])
+    fgd_df = fgd_df.drop_duplicates()
+
+    # Replace the EIA923 NA value ('.') with a real NA value.
+    fgd_df = pudl.helpers.fix_eia_na(fgd_df)
+
+    # Convert thousands of dollars to dollars
+    fgd_df.loc[:, fgd_df.columns.str.endswith("_1000_dollars")] = (
+        fgd_df.loc[:, fgd_df.columns.str.endswith("_1000_dollars")] * 1000
+    )
+    fgd_df.columns = fgd_df.columns.str.replace("_1000_dollars", "")  # Rename columns
```

Comment:
I do the same conversion in my cooling table PR, so we should probably pick one way and do it the same in both. No strong preference. I think mine uses a `for` loop so this one is probably better! I will edit mine.